WalletWasabi/JsonConverters/NetworkJsonConverter.cs

```diff
@@ -7,19 +7,13 @@ namespace WalletWasabi.JsonConverters;
 /// Converter used to convert <see cref="Network"/> to and from JSON.
 /// </summary>
 /// <seealso cref="JsonConverter" />
-public class NetworkJsonConverter : JsonConverter
+public class NetworkJsonConverter : JsonConverter<Network>
 {
 	/// <inheritdoc />
-	public override bool CanConvert(Type objectType)
-	{
-		return objectType == typeof(Network);
-	}
-
-	/// <inheritdoc />
-	public override object? ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+	public override Network? ReadJson(JsonReader reader, Type objectType, Network? existingValue, bool hasExistingValue, JsonSerializer serializer)
 	{
 		// check additional strings that are not checked by GetNetwork
-		string networkString = ((string)reader.Value).Trim();
+		var networkString = ((string?)reader.Value)?.Trim();
```

Comment:
L22 reports a warning. It can be fixed like this:

```cs
	public override Network? ReadJson(JsonReader reader, Type objectType, Network? existingValue, bool hasExistingValue, JsonSerializer serializer)
	{
		// check additional strings that are not checked by GetNetwork
		var networkString = ((string?)reader.Value)?.Trim();

		if (networkString is null)
		{
			throw new ArgumentNullException("name");
		}

		if ("regression".Equals(networkString, StringComparison.OrdinalIgnoreCase))
		{
			return Network.RegTest;
		}

		return Network.GetNetwork(networkString);
	}
```

Reply:
Done.